client/src/reducers/controls.js

```diff
@@ -74,9 +75,12 @@ const Controls = (
       };
     }
     case "reset World to eq Universe": {
+      const { userDefinedGenes, diffexpGenes } = state;
       return {
         ...state,
-        resettingInterface: false
+        resettingInterface: false,
+        userDefinedGenes: [].concat(userDefinedGenes, diffexpGenes).slice(0,globals.maxUserDefinedGenes),
```

Comment:
this removes the most recently added genes -- is that intentional?   I would have imagined we would remove the least recently added genes.

@liaprins-czi - thoughts?

Reply:
Yes I agree it should keep the most recently added and sacrifice the "oldest" genes instead... good catch

Reply:
Good points. @bkmartinjr I've implemented the solution you mentioned below: don't throw out any user-added genes and have a max gene limit that only slices off diffexp genes.